pkgs/applications/networking/instant-messengers/mikutter/update.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p bundler bundix curl jq common-updater-scripts
+# shellcheck shell=bash
+
+set -euxo pipefail
+
+main() {
+    local currentVer="$1"
+    local scriptDir="$2"
+    local latestVer
+    local srcDir
+
+    if [[ -z "$UPDATE_NIX_ATTR_PATH" ]]; then
+        echo "[ERROR] Please run the following instead:" >&2
+        echo >&2
+        echo "    % nix-shell maintainers/scripts/update.nix --argstr path mikutter" >&2
+        exit 1
+    fi
+
+    latestVer="$(queryLatestVersion)"
+    if [[ "$currentVer" == "$latestVer" ]]; then
```

Comment:
```suggestion
    if [[ $currentVer == $latestVer ]]; then
```

Reply:
Dropping quotes would result in incorrect behavior in some cases because expanded variables on the right hand side of the comparison would be interpreted as a glob pattern when using the `[[` builtin. For example:

```bash
currentVer='1.0.0'
latestVer='1*'
if [[ $currentVer == $latestVer ]]; then
  echo 'up to date'
fi
```

The above script would output "up to date" because the `*` in latestVer is interpreted as a wildcard character.

Though quotes should be unnecessary for the left hand side, I still think it's a good idea to just quote it anyways because missing quotes do more harm than redundant ones when dealing with shell scripts.

Reply:
I thought I already have shellcheck remembered but I don't.